src/openms/include/OpenMS/CHEMISTRY/EnzymaticDigestion.h

```diff
@@ -112,7 +112,7 @@ namespace OpenMS
     void digestUnmodified(const StringView& sequence, std::vector<StringView>& output, Size min_length = 1, Size max_length = 0) const;
 
     /// Returns true if the fragment starting at position @p pos with length @p length within the sequence @p sequence could be generated by the current model
-    bool isValidProduct(const String& sequence, Size pos, Size length, bool ignore_missed_cleavages = true) const;
+    bool isValidProduct(const String& sequence, Size pos, Size length, bool ignore_missed_cleavages = true, bool allow_random_asp_pro_cleavage = false) const;
```

Comment:
Hmm so we have this also in the base class

Reply:
Yes unfortunately that is the only place where I had access to the needed data. To avoid this, bigger restructuring would be necessary I guees.

Reply:
@hendrikweisser any idea how we can get this hin without making the current design obsolete.